src/pages/ProductLists/ProductLists.js

```diff
@@ -1,10 +1,112 @@
 import React, { Component } from 'react';
+import ProductContainer from '../../component/ProductContainer/ProductContainer';
+import Modal from '../../component/Modal/Modal';
+import '../ProductLists/ProductLists.scss';
+
+class ProductLists extends Component {
+  constructor() {
+    super();
+    this.state = {
+      product: [],
+      newProduct: [],
+      isModalOn: false,
+    };
+  }
+
+  componentDidMount() {
+    fetch('/data/productData.json')
+      .then(res => res.json())
+      .then(info => {
+        this.setState({
+          product: info,
+          newProduct: info,
+        });
+      });
+
+    // fetch('http://10.58.5.17:8000/product/products?type=1&sort=low_price')
+    //   .then(res => res.json())
+    //   .then(info => {
+    //     console.log(info);
+    //     this.setState({
+    //       product: info.products,
+    //     });
+    //     console.log('product', this.state.product);
+    //   });
+  }
+
+  //모달창 on/off 함수 구현 시작
+
+  handleClick = () => {
+    this.setState(state => ({
+      isModalOn: !state.isModalOn,
+    }));
+  };
+
+  //filtering button functionality
+  filteringBtns = e => {
+    let product;
+
+    const { newProduct } = this.state;
+    if (e.target.value === 'all') {
+      product = newProduct;
+    } else if (e.target.value === '낮은가격순') {
+      product = newProduct
+        .filter(item => item.discounted_price)
+        .sort((a, b) => a.discounted_price - b.discounted_price);
+    } else if (e.target.value === '높은가격순') {
+      product = newProduct
+        .filter(item => item.discounted_price)
+        .sort((a, b) => b.discounted_price - a.discounted_price);
+    }
+    this.setState({
+      product: product,
+    });
+  };
 
-export class ProductLists extends Component {
   render() {
+    const { product, isModalOn } = this.state;
     return (
-      <div>
-        <h1>ProductLists</h1>
+      <div className="Container">
+        <div className="categoryTitle">
+          <p>침대</p>
+
+          <div className="filterBtn" onClick={this.handleClick}>
+            Filtering ▼
+            {isModalOn && <Modal filteringBtns={this.filteringBtns} />}
+          </div>
+        </div>
+
+        <div className="singleProduct">
+          {product.map(productInfo => {
+            const {
+              id,
+              company,
+              product_name,
+              discount_rate,
+              discounted_price,
+              star_rate,
+              review,
+              image_url,
+            } = productInfo;
+            return (
+              <ProductContainer
+                key={id}
+                id={id}
+                company={company}
+                productName={product_name}
+                discountRate={discount_rate}
+                discountedPrice={discounted_price}
+                starRate={star_rate}
+                review={review}
+                imgUrl={image_url}
+              />
+            );
+          })}
```

Comment:
이 경우에는 구조분해할당 없이 쓰는게 더 잘보이지 않을까요? 혹은 다음과 같이 쓸거 같습니다
```suggestion
        <div className="singleProduct">
          {product.map(productInfo => <ProductContainer key={productInfo.id} {...productInfo} />)}
```

Reply:
rest parameter에 대해 오늘 공부하고 나서 바로 적용해보도록 하겠습니다.
이건 완전 꿀팁이네요 와.... 저도 모르게 소리내면서 감탄했어요

Reply:
rest parameter를 사용해주면 비구조화 할당을 해주지 않아도 되는건가요?